src/client_side_request.h

```diff
@@ -39,11 +39,9 @@ class ClientHttpRequest
     CBDATA_CLASS(ClientHttpRequest);
```

Comment:
[Jenkins](https://build.squid-cache.org/job/5-pr-test/4217/COMPILER=clang,OS=debian-stable,label=docker-build-host/console):

    src/client_side_request.h:39:5: error: 'toCbdata' overrides a member function
        but is not marked 'override' [-Winconsistent-missing-override]

To fix this particular error, you can probably use this approach:

```suggestion
#if USE_ADAPTATION
    CBDATA_CHILD(ClientHttpRequest);
#else
    CBDATA_CLASS(ClientHttpRequest);
#endif
```

... but I suspect it would not be sufficient to fix other similar problems, _and_ I think the best correct long-term fix is actually different: Adding an unconditional (virtual) AsyncJob base to the ClientHttpRequest class (and adjusting the code accordingly).

However, reviewing that proper fix while surrounded by all the noise in this PR would be very difficult (for me). Thus, I think the best solution here is one of the following two:

1. Undo "override" changes for ClientHttpRequest in this PR. Do not change ClientHttpRequest parents. Merge this PR. Followup with a dedicated PR that makes ClientHttpRequest an AsyncJob child even when `USE_ADAPTATION` is false. Add "override" specs in that second PR.
2. Pause the work on this PR. Do a dedicated PR that makes ClientHttpRequest an AsyncJob child even when `USE_ADAPTATION` is false. Resume the work on this PR after merging the first one and rebasing this one.

Neither solutions imply proper conversion of ClientHttpRequest into an AsyncJob. That should be done separately. It is probably best to do that _after_ the two PRs above. I do not recommend doing that third "proper conversion" PR at this time because I believe it requires developing more AsyncJob-related expertise.

Reply:
Done in [6364001](https://github.com/squid-cache/squid/pull/1045/commits/6364001fc669630be0b94754acc730e0af75b3a1)